pkg/frontend/admin_openshiftcluster_etcdcertificaterenew.go

```diff
@@ -0,0 +1,387 @@
+	// "EtcdMembersAvailable":             operatorv1.ConditionTrue,
+	// "NodeInstallerProgressing":         operatorv1.ConditionFalse,
+	// "NodeControllerDegraded":           operatorv1.ConditionFalse,
+	// "EtcdMembersProgressing":           operatorv1.ConditionFalse,
+}
+
+var etcdOperatorConditionsExpected = map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+	configv1.OperatorAvailable:   configv1.ConditionTrue,
+	configv1.OperatorProgressing: configv1.ConditionFalse,
+	configv1.OperatorDegraded:    configv1.ConditionFalse,
+}
+
+func (f *frontend) postAdminOpenShiftClusterEtcdCertificateRenew(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	err := f._postAdminOpenShiftClusterEtcdCertificateRenew(ctx, r, log)
+
+	adminReply(log, w, nil, nil, err)
+}
+
+// validate cluster is <4.9 and etcd is in expected state
+func (e *etcdrenew) validateAndBackup(ctx context.Context, takeBackup bool) error {
+	s := []steps.Step{
+		steps.Action(e.validateEtcdOperatorControllersState),
+		steps.Action(e.validateEtcdOperatorState),
+		steps.Action(e.validateEtcdCertsExistsAndExpiry),
+	}
+	if takeBackup {
+		s = append(s,
+			steps.Action(e.backupEtcdSecrets),
+			steps.Action(e.deleteEtcdSecrets),
+		)
+	}
+
+	_, err := steps.Run(ctx, e.log, 10*time.Second, s, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *etcdrenew) isRenewed(ctx context.Context) error {
+	s := []steps.Step{
+		steps.Condition(e.isRevised, 30*time.Minute, true),
+	}
+	_, err := steps.Run(ctx, e.log, 10*time.Second, s, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (f *frontend) _postAdminOpenShiftClusterEtcdCertificateRenew(ctx context.Context, r *http.Request, log *logrus.Entry) error {
+	resourceName := chi.URLParam(r, "resourceName")
+	resourceType := chi.URLParam(r, "resourceType")
+	resourceGroupName := chi.URLParam(r, "resourceGroupName")
+
+	resourceID := strings.TrimPrefix(r.URL.Path, "/admin")
+
+	doc, err := f.dbOpenShiftClusters.Get(ctx, resourceID)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", resourceType, resourceName, resourceGroupName)
+	case err != nil:
+		return err
+	}
+
+	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+	e := &etcdrenew{
+		log:           log,
+		k:             k,
+		secretNames:   nil,
+		mode:          "renew",
+		backupSecrets: make(map[string][]byte),
+		lastRevision:  0,
+	}
+
+	if err = e.validateClusterVersion(ctx); err != nil {
+		return err
+	}
+
+	// Fetch secretNames using nodeNames
+	masterNodeNames, err := fetchNodeNames(ctx, k, log)
+	if err != nil {
+		return err
+	}
+	if len(masterNodeNames) != 3 {
```

Comment:
Is there a case where a customer might have 5 nodes? I would have this set to less than 3 maybe?

Reply:
We setup masters, no more than 3. When less than 3 we get alert and need to recover the missing one.